src/programs/metadata/accounts/Metadata.ts

```diff
@@ -216,6 +217,26 @@ export class Metadata extends Account<MetadataData> {
     ).flat();
   }
 
+  static async findDataByOwner(
+    connection: Connection,
+    owner: AnyPublicKey,
+  ): Promise<MetadataData[]> {
+    const accounts = await TokenAccount.getTokenAccountsByOwner(connection, owner);
+    const accountsWithAmount = accounts
+      .map(({ data }) => data)
+      // Only include tokens where amount equal to 1.
+      // Note: This is not the same as mint supply.
+      // NFTs by definition have supply of 1, but an account balance > 1 implies a mint supply > 1.
+      .filter(({ amount }) => amount?.eq(new BN(1)));
+
+    const metadataAddresses = await Promise.all(
+      accountsWithAmount.map(async ({ mint }) => await Metadata.getPDA(mint)),
+    );
```

Comment:
```suggestion
    const metadataPDALookups = accounts
      // Only include tokens where amount equal to 1.
      // Note: This is not the same as mint supply.
      // NFTs by definition have supply of 1, but an account balance > 1 implies a mint supply > 1.
      .reduce<Promise<T>>((memo, { data }) => {
        if (data.ammount?eq(new BN(1))) {
          return memo;
        }
        
        cont lookup = Metadata.getPDA(data.mint);
        
        return [... memo, lookup];
      }, []);
      
      const metdataAddress = await Promise.all(metadataPDALookup)
```

Possibility to get to 1 loop with doing a reduce.

Reply:
Thanks, I've implemented your suggested change but reversed the logic for the `if` statement as I believe the correct behaviour is to include when the amount is 1.